src/common/Formatter.cc

```diff
@@ -29,27 +30,39 @@ namespace ceph {
 std::string
 fixed_u_to_string(uint64_t num, int scale)
 {
-	std::ostringstream t;
+	CachedStackStringStream css;
 
-	t.fill('0');
-	t.width(scale + 1);
-	t << num;
-	int len = t.str().size();
-	return t.str().substr(0,len - scale) + "." + t.str().substr(len - scale);
+	css->fill('0');
+	css->width(scale + 1);
+	*css << num;
+	auto len = css->strv().size();
```

Comment:
I guess we call `strv()` three times. I'm not sure compilers optimize this.

Reply:
strv() is very cheap (`O(1)`) in this case so it doesn't really matter.

Reply:
:+1: